docs/tasks/istio-installation.md

```diff
@@ -7,31 +7,105 @@ layout: docs
 
-1. Do this.
-1. Do this next. Possibly read this [related explanation](...).
+1. Download and extract the [istio installation files](https://raw.githubusercontent.com/istio/istio/master/releases/istio-alpha.tar.gz), or
+clone the github [istio](https://github.com/istio/istio) repository:
+```bash
+git clone https://github.com/istio/istio
+```
+1. Change directory to istio:
+```bash
+cd istio
+```
+2. Install the Istio core components (Istio-Manager, Mixer, and Ingress-Controller):
+```bash
+kubectl apply -f ./kubernetes/istio.yaml
+```
+1. Source the Istio configuration file:
+```bash
+source istio.VERSION
+```
+3. Download one of the [`istioctl`]({{site.bareurl}}/docs/reference/istioctl.html) client binaries corresponding to your OS: `istioctl-osx`, `istioctl-win.exe`,
+`istioctl-linux`, targeted at Mac, Windows or Linux users respectively. For example, run the following commands on a Mac system:
+```bash
+curl ${ISTIOCTL_URL}/istioctl-osx > /usr/local/bin/istioctl
+chmod +x /usr/local/bin/istioctl
+```
+Istioctl is needed to inject Envoy as a sidecar proxy. It also provides a convenient CLI for creating routing rules and policies.
+4. Deploy your application with Envoy:
+```bash
+kubectl create -f <(istioctl kube-inject -f <your-app-spec>.yaml)
+```
+The [kube-inject]({{site.bareurl}}/docs/reference/istioctl.html##kube-inject) tool will automatically inject an Envoy container in the pod running the application.
+Alternatively, deploy one of the samples applications, for instance [bookinfo]({{site.bareurl}}/docs/samples/bookinfo.html).
+5. Optionally: to view metrics collected by Mixer, install [Prometheus](https://prometheus.io), [Grafana](http://staging.grafana.org) and ServiceGraph addons:
+```bash
+kubectl apply -f ./kubernetes/addons/grafana.yaml
+kubectl apply -f ./kubernetes/addons/prometheus.yaml
+kubectl apply -f ./kubernetes/addons/servicegraph.yaml
+```
+The Grafana image provided as part of this sample contains a built-in Istio dashboard that you can access from:
+```bash
+http://<grafana-svc-external-IP>:3000/dashboard/db/istio-dashboard
+```
+The addons yaml files contain services configured as type LoadBalancer. If services are deployed with type NodePort,
+start kubectl proxy, and edit Grafana's Istio-dashboard to use the proxy. Access Grafana via kubectl proxy:
+```bash
+http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/grafana:3000/dashboard/db/istio-dashboard
+```
 {% endcapture %}
 
 {% capture discussion %}
-## Understanding ...
+## Verify the installation
+
+1. Ensure the following Kubernetes services were deployed: "istio-manager", "istio-mixer", and "istio-ingress-controller".
+```bash
+kubectl get svc
+NAME                       CLUSTER-IP     EXTERNAL-IP     PORT(S)              AGE
+istio-ingress-controller   10.83.241.84   35.184.70.168   80:30583/TCP         39m
```

Comment:
can we rename this to `istio-ingress`? it's no longer a controller, just a dumb proxy.